### ISSUE TITLE
Backprop: Reduce generational dampening that slows training convergence

### ISSUE THREAD

**Issue by stservice** [stSoftwareAU/NEAT-TS#1436]
## Problem

The backpropagation weight and bias averaging in `src/propagate/Weight.ts` and `src/propagate/Bias.ts` blends accumulated gradients with `config.generations` copies of the original weight/bias value:

```typescript
// Weight.ts
const generations = config.generations + cs.count - totalActivationCount;
const totalGenerationalWeight = c.weight * generations;
const averageWeight = (synapseAverageWeightTotal + totalGenerationalWeight) / (totalActivationCount + generations);
```

When `config.generations` is high (default can be random 1-100), this creates **aggressive weight inertia** — the network is biased towards its original weights and slow to adapt. Combined with the default learning rate of `random()³` (biased towards very small values, mean ~0.05), training can appear "stuck" or converge extremely slowly.

This may explain why backpropagation underperforms compared to memetic evolution and fine-tuning, which use bounded adaptive steps without this generational dampening.

## Suggested Improvement

1. Make generational dampening optional or configurable
2. Consider decaying the generational factor over training iterations:
   ```typescript
   const generationalFactor = Math.max(0, 1 - (iteration / totalIterations));
   const effectiveGenerations = config.generations * generationalFactor;
   ```
3. Alternatively, cap `generations` at a reasonable multiple of `count` to prevent excessive dampening

## Files

- `src/propagate/Weight.ts` (lines ~109-113)
- `src/propagate/Bias.ts` (lines ~86-92)
- `src/propagate/BackPropagation.ts` (learning rate defaults, lines ~101-109)

## Expected Outcome

Faster convergence in backpropagation training, bringing its effectiveness closer to memetic evolution for standard tasks.
